org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/antlr/internal/AbstractInternalContentAssistParser.java

```diff
@@ -7,530 +7,116 @@
  *******************************************************************************/
 package org.eclipse.xtext.ide.editor.contentassist.antlr.internal;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-import org.antlr.runtime.BitSet;
-import org.antlr.runtime.DFA;
-import org.antlr.runtime.FailedPredicateException;
-import org.antlr.runtime.IntStream;
-import org.antlr.runtime.Parser;
-import org.antlr.runtime.RecognitionException;
 import org.antlr.runtime.RecognizerSharedState;
 import org.antlr.runtime.Token;
 import org.antlr.runtime.TokenStream;
-import org.apache.log4j.Logger;
-import org.eclipse.emf.ecore.EObject;
-import org.eclipse.emf.ecore.util.EcoreUtil;
-import org.eclipse.xtext.AbstractElement;
 import org.eclipse.xtext.Grammar;
 import org.eclipse.xtext.GrammarUtil;
 import org.eclipse.xtext.TerminalRule;
-import org.eclipse.xtext.UnorderedGroup;
 import org.eclipse.xtext.ide.editor.contentassist.antlr.FollowElement;
 import org.eclipse.xtext.ide.editor.contentassist.antlr.LookAheadTerminal;
 import org.eclipse.xtext.ide.editor.contentassist.antlr.LookAheadTerminalRuleCall;
 import org.eclipse.xtext.ide.editor.contentassist.antlr.LookaheadKeyword;
 import org.eclipse.xtext.ide.editor.contentassist.antlr.ObservableXtextTokenStream;
-import org.eclipse.xtext.parser.antlr.ITokenDefProvider;
-import org.eclipse.xtext.parser.antlr.IUnorderedGroupHelper;
 import org.eclipse.xtext.parser.antlr.TokenTool;
 
-import com.google.common.collect.Iterators;
-import com.google.common.collect.LinkedHashMultimap;
-import com.google.common.collect.Lists;
-import com.google.common.collect.Maps;
-import com.google.common.collect.Multimap;
-
-/*
- * Initially copied from org.eclipse.xtext.ui.editor.contentassist.antlr.internal.AbstractInternalContentAssistParser
- */
 /**
  * @since 2.9
  */
-public abstract class AbstractInternalContentAssistParser extends Parser implements
-		ObservableXtextTokenStream.StreamListener, ITokenDefProvider {
+public abstract class AbstractInternalContentAssistParser extends BaseInternalContentAssistParser<FollowElement, LookAheadTerminal> implements
+		ObservableXtextTokenStream.StreamListener {
+
+	protected class DefaultFollowElementFactory extends AbstractFollowElementFactory<FollowElement, LookAheadTerminal> implements IFollowElementFactory {
+
+		protected DefaultFollowElementFactory() {
```

Comment:
does this need a since? is this a api change?

Reply:
You mean because the constructor is no longer public in the protected inner class?

Reply:
But yes, that is the question that I would have hoped for, the API tooling could answer.

Reply:
yes thats what i meant